### PR TITLE
Improve over 10% on lmbench-stat/open/fs by refactoring `Dentry`

### PR DESCRIPTION
**Pull request by lucassong-mh**
The key change is we shouldnot do heap allocation every time we create a `Dentry`.

### CODE REVIEW

**Comment by lucassong-mh** [comment]
@boterinas ready
@boterinas label O-test

**Comment by Plucky923** [comment]
LGTM！ In the system, `dentry` is frequently accessed, and using `Arc` will cause dentry to be allocated from the heap, which will affect efficiency. I think this contribution will greatly improve the efficiency of the system.

**Comment by Plucky923** [comment]
LGTM！ In the system, `dentry` is frequently accessed, and using `Arc` will cause dentry to be allocated from the heap, which will affect efficiency. I think this contribution will greatly improve the efficiency of the system.

File: kernel/src/fs/path/dentry.rs [hunk 1]
```diff
@@ -139,129 +138,100 @@ impl Dentry_ {
...
+        self.children.check_mountpoint(name)?;
 
-        let children = self.children.upread();
-        let _ = children.find_dentry_with_checking_mountpoint(name)?;
-        self.inode.unlink(name)?;
-
-        let mut children = children.upgrade();
-        children.delete_dentry(name);
+        self.inode().unlink(name)?;
+        self.children.delete(name);
```
[the remark this answers is not in the document]
Comment (lucassong-mh) [reply kernel/src/fs/path/dentry.rs]: Make sense and revised

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**